### PR TITLE
MRG: fix dprime_2afc

### LINKED ISSUE #347: bug in dprime_2afc
apparently we borked the `dprime_2afc` function.  There's been discussion on slack as to whether we should try to fix it, or just scuttle it.

### What is wrong

1. computation of `dprime_2afc` takes number of "correct" and "incorrect" trials, and passes those to `dprime` as "hits" and "false alarms".  Why we did that is a bit of a mystery, but obviously it makes it impossible to compute hit *rate* (`H/(H+M)`) and false alarm *rate* (`F/(F+C)`) if we're implicitly asking users to lump H and C into "correct" and lump M and F into "incorrect".

2. techinically, a 2AFC task is easier than a yes-no (signal present/absent) task.  Macmillan & Creelman (p 166-168) show that a correction factor of `1 / sqrt(2)` is appropriate for computing dprime in 2AFC tasks.  We are not doing this correction.

### how to fix it

1. scuttle the `dprime_2afc` function, and let users do `dprime(...) / np.sqrt(2)` manually if they know they're doing 2AFC.

2. scuttle `dprime_2afc` and add a flag `dprime(..., tafc=False)` that, when `True`, adds the correction factor.

3. fix the `dprime_2afc` function to take in *hit rate* (`H/(H+M)`) and *false alarm rate* (`F/(F+C)`) instead of number of correct / incorrect trials, and include the correction factor.

cc @rkmaddox @larsoner

### PR DESCRIPTION
**Pull request by drammock**
closes #347

### CODE REVIEW

File: expyfun/analyze/_analyze.py [hunk 1]
```diff
@@ -347,64 +350,42 @@ def dprime(hmfc, zero_correction=True, return_bias=False):
...
+    standard normal distribution (AKA the percent point function). For two-
+    interval two-alternative forced-choice tasks, Macmillan & Creelman [1]_
+    recommend modifying the d′ calculation to account for the presentation of
+    two stimuli per trial (one in each interval) by dividing by ``sqrt(2)``::
+
+        (Z(hit_rate) - Z(false_alarm_rate)) / sqrt(2)
+
+    References
+    ----------
+    Macmillan & Creelman (2005). Detection theory: a user’s guide (2nd ed.).
```
Comment (larsoner) [inline expyfun/analyze/_analyze.py]: use `.. [1] Macmillan ...` here to link properly to your `[1]_` above

**Comment by drammock** [comment]
@larsoner this should be ready for review (sorry about the repeated travis fails; I'm traveling and my testing setup on the laptop is not stellar)

**Comment by larsoner** [comment]
In lieu of nitpick comments I pushed two small commits, will merge once the CIs come back happy

**Comment by larsoner** [comment]
AppVeyor is just a timeout.

@rkmaddox did you want to look?

File: examples/analysis/analysis_demo.py [hunk 2]
```diff
@@ -5,7 +5,8 @@
 =============
 
 This example simulates some 2AFC data and demonstrates the analysis
-functions ``dprime()`` and ``barplot()``.
+functions :func:`expyfun.analyze.dprime` and
+:func:`expyfun.analyze.barplot()`.
```
Comment (drammock) [inline examples/analysis/analysis_demo.py]: does it matter that one has parentheses and the other doesn't?  (not trying to bust your chops... just want to get it right myself next time)

File: expyfun/analyze/_analyze.py [hunk 3]
```diff
@@ -361,7 +361,7 @@ def dprime(hmfc, zero_correction=True, return_bias=False, two_interval=False):
     References
     ----------
     .. [1] Macmillan & Creelman (2005). Detection theory: a user's guide (2nd
-    ed.). Mahwah, NJ: Lawrence Erlbaum Associates, pp. 166-170.
+           ed.). Mahwah, NJ: Lawrence Erlbaum Associates, pp. 166-170.
```
Comment (drammock) [inline expyfun/analyze/_analyze.py]: ah... I wondered if I was supposed to indent that.  thanks for fixing

File: examples/analysis/analysis_demo.py (hunk 2 above)
Comment (larsoner) [reply examples/analysis/analysis_demo.py]: whoops! apparently not, CircleCI is happy enough:

https://322-11614571-gh.circle-artifacts.com/0/home/ubuntu/expyfun/doc/_build/html/auto_examples/analysis/analysis_demo.html

**Comment by larsoner** [comment]
Thanks @drammock

**Pull request merged**